core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -372,15 +372,17 @@ object SparkEnv extends Logging {
 
     val useLegacyMemoryManager = conf.getBoolean("spark.memory.useLegacyMode", false)
     val memoryManager: MemoryManager =
-      conf.getOption("spark.memory.manager").filterNot(_.equalsIgnoreCase("default"))
-          .map(Utils.classForName(_)
-          .getConstructor(classOf[SparkConf], classOf[Int])
-          .newInstance(conf, Int.box(numUsableCores))
-          .asInstanceOf[MemoryManager]).getOrElse {
-        if (useLegacyMemoryManager) {
-          new StaticMemoryManager(conf, numUsableCores)
-        } else {
-          UnifiedMemoryManager(conf, numUsableCores)
+      SparkSnappyUtils.loadSnappyManager(conf, numUsableCores).getOrElse {
+        conf.getOption("spark.memory.manager").filterNot(_.equalsIgnoreCase("default"))
+            .map(Utils.classForName(_)
+                .getConstructor(classOf[SparkConf], classOf[Int])
+                .newInstance(conf, Int.box(numUsableCores))
+                .asInstanceOf[MemoryManager]).getOrElse {
+          if (useLegacyMemoryManager) {
+            new StaticMemoryManager(conf, numUsableCores)
+          } else {
+            UnifiedMemoryManager(conf, numUsableCores)
+          }
```

Comment:
An explicit setting of "spark.memory.manager" should override SnappyUMM. So this should be changed in the getOrElse block of previous code:

```scala
      conf.getOption("spark.memory.manager").filterNot(_.equalsIgnoreCase("default"))
      ...
          .asInstanceOf[MemoryManager]).getOrElse {
            SparkSnappyUtils.loadSnappyManager(conf, numUsableCores).getOrElse {
              if (useLegacyMemoryManager)
              ...
            }
          }
```